src/providers/goDaddy.ts

```diff
@@ -0,0 +1,17 @@
+import fetch from '../helpers/httpRequest'
+import { getData } from '../api/lib/data'
+import { ServiceKey } from '../api/types/admin'
+
+export const getDomains = () => {
+  const serviceKeys: any = getData('serviceKeys')
+  const { service, value: key } = serviceKeys.find(
+    ((el: ServiceKey ) => el.key === 'GD_Key'
+  ))
+  const { value: secret } = serviceKeys.find(
+    (el: ServiceKey) => el.key === 'GD_Secret'
+  )
+  const url = `${service}/v1/domains?statuses=ACTIVE`
+  const options = { headers: { Authorization: `sso-key ${key}:${secret}` } }
+
+  return fetch(url, options)
```

Comment:
question: should this be `fetch(url, options).then(r => r.json())`

Reply:
Ho maybe `fetch` is not a good name I did a helper function from `src/helpers/httpRequest.ts` with `async/await`

```
import fetch from 'node-fetch'

export default async (url: string, options: object) => {
  try {
    const request = await fetch(url, options)
    return await request.json()
  } catch (error) {
    throw new Error(error)
  }
}
```

Reply:
yeah fetch is not a good name, haha. It's going to be confusing for future developers. Please take care of this later. https://github.com/garageScript/myProxy/issues/47